### DISCUSSION TITLE
OTA process documentation

### DISCUSSION THREAD

**Discussion by Skostrub** [devbis/z03mmc#27] [Q&A]
No documentation for OTA update provided.

Zigbee Firmware assumes to use this device with zigbee2mqtt package in home automation systems, but this unsupported (yet I hope) device require some configuration and manual setup to be OTA updated through zigbee2mqtt.

**Comment by devbis** [comment]
The firmware supports generic OTA updates. See OTA documentation for coordinator software you use.
https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files for z2m local updates or remote OTA index from zigbee2mqtt

Feel free to add missing paragraphs in the documentation with a PR

**Comment by sv1kwz** [comment]
any documentation for OTA update with ZHA  ?

**Reply by morenod** [reply]
I have just seen the new release 1.1.0, I would like to upgrade to it.

@devbis if you have any guide about how to upgrade on ZHA, I will use it and I will PR it here on the docs section

**Reply by devbis** [reply]
There is an "official" documentation from zigpy (ZHA underlying library) https://github.com/zigpy/zigpy/wiki/OTA-Device-Firmware-Updates

I haven't tried it. Also, I don't know if ZHA looks for z2m repository for OTA. In this case, wait https://github.com/Koenkk/zigbee-OTA/pull/440 to merge.

**Comment by veulsan** [comment]
Read [this,](https://github.com/zigpy/zigpy/pull/1340) on zigbpy and figured out how to make Home Assistant do OTA updates with files from a directory.

In configuration.yaml add the directory and download your image file to the directory:
zha:
  zigpy_config:
    advanced_ota_dir: /config/zigbee_ota

It should afterwards look like this under Settings :)
![image](https://github.com/devbis/z03mmc/assets/144459539/41495b62-61e7-4a4f-a51b-1d4a25a7e658)

**Comment by JGeeforce** [comment]
I spent several hours to use HA latest version to rollback from zigbee to BT, but without success. I created the folder, placed firmware (.zigbee) files in it, restarted HA. ANd tried to trigger the update, but then the firmware version goes from a known version to unknown and nothing happens. 

Anyone managed to go back to BT and has a step by step instrucion?
